### PR TITLE
rgw,test: Add even more DoutPrefixProvider's to the RGW codebase

### PR DESCRIPTION
**Pull request by alimaredia**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by alimaredia** [comment]
Passing full rgw teuthology suite here: https://pulpito.ceph.com/amaredia-2021-03-30_01:06:23-rgw-wip-rgw-dpp-update-master-distro-basic-smithi/

**Review by dang — APPROVED** [review]

**Comment by alimaredia** [comment]
@cbodley @dang This was my latest run on teuthology: https://pulpito.ceph.com/amaredia-2021-04-15_19:59:33-rgw-wip-rgw-dpp-update-master-distro-basic-smithi/ No crashes but many failed tests for reasons that don't seem related to the PR?

**Comment by alimaredia** [comment]
jenkins test api

**Comment by alimaredia** [comment]
https://pulpito.ceph.com/amaredia-2021-04-28_22:49:00-rgw-wip-rgw-dpp-update-master-distro-basic-smithi/ for a clean teuthology run

**Comment by yuvalif** [comment]
why adding non-null DPPs to test files?
i don't think we have logs in these

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -7525,7 +7527,7 @@ int RGWHandler::init(rgw::sal::Store* _store,
 
 int RGWHandler::do_init_permissions(const DoutPrefixProvider *dpp, optional_yield y)
 {
-  int ret = rgw_build_bucket_policies(dpp, store, s, y);
+  int ret = rgw_build_bucket_policies(s, store, s, y);
```
Comment (dang) [inline src/rgw/rgw_op.cc:7530]: Should this one be dpp and not s?

**Review by dang — APPROVED** [review]
Still looks fine.

File: src/rgw/rgw_op.cc (hunk 1 above)
Comment (alimaredia) [reply src/rgw/rgw_op.cc:7530]: Yes, I'll make a note to change this when I go ahead and squash the commits.

**Review by ivancich — APPROVED** [review]

**Comment by alimaredia** [comment]
clean teuthology run after rebase, and commit squash: https://pulpito.ceph.com/amaredia-2021-05-05_04:17:30-rgw-wip-rgw-dpp-update-master-distro-basic-smithi/

**Pull request merged**